programs/server/play.html

```diff
@@ -559,6 +598,51 @@
         document.getElementById('error').style.display = 'block';
     }
 
+    /// Huge JS libraries should be loaded only if needed.
+    function loadJS(src) {
+        return new Promise((resolve, reject) => {
+            const script = document.createElement('script');
+            script.src = src;
+            script.addEventListener('load', function() { resolve(true); });
+            document.head.appendChild(script);
+        });
+    }
+
+    let load_dagre_promise;
+    function loadDagre() {
+        if (load_dagre_promise) { return load_dagre_promise; }
+
+        load_dagre_promise = Promise.all([
+            loadJS('https://dagrejs.github.io/project/dagre/latest/dagre.min.js'),
+            loadJS('https://dagrejs.github.io/project/graphlib-dot/latest/graphlib-dot.min.js'),
+            loadJS('https://dagrejs.github.io/project/dagre-d3/latest/dagre-d3.min.js'),
+            loadJS('https://cdn.jsdelivr.net/npm/d3@7'),
```

Comment:
Maybe it's better to specify version not to break script if `latest` will have incompatible changes?
```suggestion
            loadJS('https://dagrejs.github.io/project/dagre/v0.8.5/dagre.min.js'),
            loadJS('https://dagrejs.github.io/project/graphlib-dot/v0.6.4/graphlib-dot.min.js'),
            loadJS('https://dagrejs.github.io/project/dagre-d3/v0.6.4/dagre-d3.min.js'),
            loadJS('https://cdn.jsdelivr.net/npm/d3@7.0.0'),
```